### PR TITLE
fix: refactor code to use native promises instead of q

### PR DESCRIPTION
**Pull request by armano2**
Native promises are supported by node for some time, and there is no point in using external library that provides "same" functionality as native function

https://github.com/kriskowal/q/issues/843
https://github.com/kriskowal/q/issues/827#issuecomment-366897835

-------------------------

https://node.green/#ES2015-built-ins-Promise

### CODE REVIEW

File: packages/conventional-changelog-core/lib/merge-config.js [hunk 1]
```diff
@@ -94,24 +94,35 @@ function mergeConfig (options, context, gitRawCommitsOpts, parserOpts, writerOpt
...
+      pkgPromise = readPkgUp()
     }
   }
 
-  const gitRemoteOriginUrlPromise = Q(gitRemoteOriginUrl())
+  const gitRemoteOriginUrlPromise = gitRemoteOriginUrl()
 
-  return Q.allSettled([configPromise, pkgPromise, semverTagsPromise(options), gitRemoteOriginUrlPromise])
-    .spread(function (configObj, pkgObj, tagsObj, gitRemoteOriginUrlObj) {
+  // TODO: replace with Promise.allSettled for node 12.10+
```
Comment (armano2) [inline packages/conventional-changelog-core/lib/merge-config.js:113]: https://node.green/#ES2020-features--Promise-allSettled

File: packages/conventional-changelog-core/lib/merge-config.js [hunk 2]
```diff
@@ -19,6 +19,17 @@ const _ = require('lodash')
 
 const rhosts = /github|bitbucket|gitlab/i
 
+// TODO: delete when min node target is set to 12.10+
```
Comment (armano2) [inline packages/conventional-changelog-core/lib/merge-config.js:22]: https://node.green/#ES2020-features--Promise-allSettled

File: packages/conventional-changelog-core/lib/merge-config.js [hunk 3]
```diff
@@ -19,6 +19,17 @@ const _ = require('lodash')
...
+Promise.allSettled = Promise.allSettled || function (promises) {
+  const wrappedPromises = promises.map((p) =>
+    Promise.resolve(p).then(
+      (value) => ({ status: 'fulfilled', value }),
+      (reason) => ({ status: 'rejected', reason })
+    )
+  )
+  return Promise.all(wrappedPromises)
+}
```
Comment (armano2) [inline packages/conventional-changelog-core/lib/merge-config.js:31]: will try to use native function if available, else apply polyfill

https://github.com/kriskowal/q/blob/01252cbe917178f5935cbe2a1ab43e12934c7057/q.js#L1693-L1703


https://node.green/#ES2020-features--Promise-allSettled

**Comment by jimmywarting** [comment]
any progress on this? 
Seems to be a bit out-of-date...

File: packages/conventional-changelog-conventionalcommits/writer-opts.js [hunk 4]
```diff
@@ -2,8 +2,8 @@
 
 const addBangNotes = require('./add-bang-notes')
 const compareFunc = require('compare-func')
-const Q = require('q')
-const readFile = Q.denodeify(require('fs').readFile)
+const promisify = require('util').promisify
+const readFile = promisify(require('fs').readFile)
```
Comment (armano2) [inline packages/conventional-changelog-conventionalcommits/writer-opts.js:6]: depends on node target we could update this to
```
require('fs').promises.readFile
``

**Comment by armano2** [comment]
this change seem to be no longer needed as #974 has been merged

**Pull request closed without merging**